spec/controllers/blocks_controller_spec.rb

```diff
@@ -0,0 +1,198 @@
+require 'rails_helper'
+
+RSpec.describe BlocksController, type: :controller do
+  describe "GET index" do
+    it "requires login" do
+      get :index
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "succeeds" do
+      login
+      get :index
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe "GET new" do
+    it "requires login" do
+      get :new
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "succeeds" do
+      login
+      get :new
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe "POST create" do
+    it "requires login" do
+      post :create
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "requires a blocked user" do
+      login
+      post :create
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+    end
+
+    it "requires valid variables" do
+      login
+      post :create, params: { block: { blocked_user_id: create(:user).id, hide_me: -1 } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+    end
+
+    it "requires an existing blocked user" do
+      login
+      post :create, params: { block: { blocked_user_id: ([1] + User.order(id: :desc).limit(1).pluck(:id)).max + 1 } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+      expect(assigns[:block].blocked_user).to be_nil
+    end
+
+    it "requires users not be the same" do
+      user = create(:user)
+      login_as(user)
+      post :create, params: { block: { blocked_user_id: user.id } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+      expect(flash[:error][:array]).to include("User cannot block themself")
+    end
+
+    it "succeeds" do
+      login
+      post :create, params: { block: { blocked_user_id: create(:user).id } }
+      expect(response).to redirect_to(blocks_url)
+    end
+
+    it "sets expected variables" do
```

Comment:
Is this on a failure? If so, what's the failure? We should be verifying that it fails, if it does.